### ISSUE TITLE
0.8.0 fails to launch

### ISSUE THREAD

**Issue by 0323pin** [tobealive/wthrr-the-weathercrab#61]
Hi,

I was a few days late and pushed 0.7.0 into our repositories today and all was working fine.
Now, I've just built 0.8.0 but I'm getting

```
Error: No such file or directory (os error 2)
Exception: wthrr exited with 1
```

That's all.
Reverted to 0.7.0 and it's working. Any clues?
I'm currently building my whole system from source, no pre-compiled binaries and still don't have a web-browser. Besides, I'll be away from Thursday until next Tuesday.

Just an heads-up in case it takes sometime to reply :)

**Comment by tobealive** [comment]
Hey @0323pin 
It's good to get report on such errors. Since I'm limited resource wise it's not always possible for me to spot them. So thanks alot for your report, it really helps to improve the app!

Most probably I missed a clause when checking for the translations that can now be saved to a file.

When I have I free moment I'll investigate later this day else it'll be fixed tomorrow 👍

**Comment by 0323pin** [comment]
Cheers.
The error itself doesn't say much, hopefully it will be easy to spot. Thanks!

**Comment by tobealive** [comment]
I can only reproduce this error if I delete the `locales/` directory in the root of the repo. I should probably make sure in `Cargo.toml` that it is a dependency. I would like to keep it as a build file and save the users from having to fetch it or to have it stored on their machine post build. What is you thought on that?

And can you verify that building is possible when using the full repo?

**Comment by 0323pin** [comment]
@tobealive I see you have edited your original reply, it differs from the automatic e-mail github sends out.

> And can you verify that building is possible when using the full repo?

We do fetch the full repo to build the package. But, then we/I do is,
```
${INSTALL_PROGRAM} ${WRKSRC}/target/release/wthrr \
		${DESTDIR}${PREFIX}/bin
```

You can find the full package Makefile [here](http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/misc/wthrr/Makefile?rev=1.13&content-type=text/x-cvsweb-markup), rust dependencies are declared in a separate [cargo-depends.mk](http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/misc/wthrr/cargo-depends.mk?rev=1.10&content-type=text/x-cvsweb-markup) file which, is called from the Makefile with `.include "cargo-depends.mk"`

Do you mean I should fetch and install `locales/pure-rust-locales.txt` somewhere in the root file system? In this case, where should it be located?

> I definitely need to get the dependencies in Cargo.toml straight.

Let's try that, else it's possible to fetch it from the build directory and install it as a separate file. I usually do this with, _e.g._ `.desktop` files.

**Comment by tobealive** [comment]
Yes, after realizing the silliness of this mistake, I changed the initial answer to reflect that this definitely needs to be fixed.

> Do you mean I should fetch and install locales/pure-rust-locales.txt somewhere in the root file system? In this case, where should it be located?

Nope, that's not wanted.

---

I included it as a constant now which should make sure it's included in the build process. As far as I could check this did the trick.

**Comment by 0323pin** [comment]
@tobealive I've just built 0.8.2 locally and everything is working fine again, thanks!

Didn't have the time to merge the update yet. Need to get some sleep. I'll try to merge it in the morning, else Tuesday  late evening. Closing this now.

**Comment by 0323pin** [comment]
Done, https://mail-index.netbsd.org/pkgsrc-changes/2023/02/23/msg269887.html

**Comment by tobealive** [comment]
Awesome :muscle: thank you 😊!

**Comment by 0323pin** [comment]
@ttytm Long time, no see. Sorry for posting on a closed thread but, I was just wondering if a new release is due anytime soon? 76 commits after latest! Any breaking stuff?

Thanks and regards.

**Comment by ttytm** [comment]
Hey @0323pin, no worries, good to hear from you!

Yes it's planned. The wasn't happening much in the last time here.
The changes have been mostly small refactoring and dependency updates. 
Next thing I'll work on is finishing the fix for #135 then a new version is ripe.

**Comment by 0323pin** [comment]
Nice, thanks!

I'll keep an eye on it. Hopefully, I'll remember to switch `MASTER_SITES=	${MASTER_SITE_GITHUB:=tobealive/}` to your new github repo :smile:

**Comment by kevinmatthes** [comment]
@ttytm @0323pin

Regarding the question of a new version being published, I created a poll (#248) on my idea to release at least a patch every month in order to benefit from the changes of the latest dependency updates.  Maybe we could continue the discussion of the next release's contents there, as well.
